services/ui/ws/window_tree.cc

```diff
@@ -217,6 +217,25 @@ void WindowTree::DoOnEmbed(mojom::WindowTreePtr tree,
 
 void WindowTree::OnAcceleratedWidgetAvailableForDisplay(Display* display) {
   DCHECK(window_manager_internal_);
+
+  if (window_server_->IsInExternalWindowMode()) {
+    ClientWindowId client_window_id;
+    if (pending_client_window_id_ != kInvalidClientId) {
+      client_window_id = MakeClientWindowId(pending_client_window_id_);
+    } else {
+      auto* display_root = display->GetWindowManagerDisplayRootForUser(
+          window_manager_state()->user_id());
```

Comment:
I have a small concern here:

167 const WindowManagerDisplayRoot* Display::GetWindowManagerDisplayRootForUser(
168     const UserId& user_id) const {
169   auto iter = window_manager_display_root_map_.find(user_id);
170   return iter == window_manager_display_root_map_.end() ? nullptr
171                                                         : iter->second;
172 }
173 
174 const WindowManagerDisplayRoot* Display::GetActiveWindowManagerDisplayRoot()
175     const {
176   return GetWindowManagerDisplayRootForUser(
177       window_server_->user_id_tracker()->active_id());

... ws::Display calls out to WindowServer to query the WindowManagerDisplayRoot here, with a given ID.

However, we create all WindowManagerDisplayRoot's with the same ID. So, I guess it only works for the first ws::WindowManagerDisplayRoot created?

Maybe it is fine, because we only need to call this method one?

Reply:
Right, this method is called only once.

Reply:
Well, I'm removing this part of the code and will always use pending client id, because it's guaranteed it's valid when we receive an accelerated widget first.